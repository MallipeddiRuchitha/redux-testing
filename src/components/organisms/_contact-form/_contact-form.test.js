import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from './_contact-form';
//import {render,fireEvent} from '../../../test-utils'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
 describe("ContactForm component",() =>{
    const store = createStore(() => ({ contacts: [] }))
    test("Matches the snapshot",() => {
        const input = create(<Provider store={store}><ContactForm /></Provider>);
        expect(input.toJSON()).toMatchSnapshot();

    })
    test("Contact element is rendered",() => {
        const { getByTestId, getByText } = render(<Provider store={store}><ContactForm/></Provider>);
        expect(getByTestId("contactform")).toBeInTheDocument();
        
    });
   
   
   

});