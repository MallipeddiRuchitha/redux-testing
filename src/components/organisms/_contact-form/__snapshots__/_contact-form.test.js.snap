// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleContactMainContent-1"
  data-testid="contactform"
>
  <div
    className="makeStyles-forms-3"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
       Contacts Application
    </h4>
    <h5
      className="MuiTypography-root makeStyles-heading-5 MuiTypography-h5"
    >
      Add Contact Form
    </h5>
    <form>
      <div
        className="makeStyles-input-4"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
        >
          Contact Name
        </h6>
        <div
          className="makeStyles-styleInputFields-8"
          data-testid="input"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="value"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <h6
          className="MuiTypography-root MuiTypography-h6"
        >
          Email
        </h6>
        <div
          className="makeStyles-styleInputFields-8"
          data-testid="input"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            onClick={[Function]}
            required={true}
            style={
              Object {
                "color": "#000",
              }
            }
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              data-testid="value"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="makeStyles-button-6"
        >
          <div
            data-testid="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiTypography-root MuiTypography-button"
                >
                   
                  Save
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
  <hr
    className="MuiDivider-root"
  />
</div>
`;
